src/modules/billing/mappers/charge-module-decorators.js

```diff
@@ -47,16 +59,22 @@ const decorateInvoice = (invoice, cmResponse) => {
   validators.assertObject(cmResponse);
 
   // Set invoice totals
-  invoice.totals = totalsMapper.chargeModuleBillRunToInvoiceModel(cmResponse, invoice.invoiceAccount.accountNumber, invoice.financialYear.yearEnding);
+  invoice.totals = mappers.totals.chargeModuleBillRunToInvoiceModel(cmResponse, invoice.invoiceAccount.accountNumber, invoice.financialYear.yearEnding);
 
   // Set de-minimis flag
   const customerFinYearSummary = findCustomerFinancialYearSummary(cmResponse, invoice.invoiceAccount.accountNumber, invoice.financialYear.endYear);
   invoice.isDeMinimis = customerFinYearSummary.deminimis;
 
-  // Decorate each invoice transaction
-  const transactionMap = indexChargeModuleTransactions(customerFinYearSummary.transactions);
+  const transactionsByLicence = indexTransactionsByLicenceNumber(customerFinYearSummary.transactions);
+  // Decorate each invoice transaction and add missing transactions created in the CM
   invoice.invoiceLicences.map(invoiceLicence => {
-    invoiceLicence.transactions.map(transaction => decorateTransaction(transaction, transactionMap));
+    transactionsByLicence.get(invoiceLicence.licence.licenceNumber).map(cmTransaction => {
+      const serviceTransaction = invoiceLicence.transactions.find(trans => trans.externalId === cmTransaction.id);
+      if (serviceTransaction) {
+        return decorateTransaction(serviceTransaction, cmTransaction);
+      }
```

Comment:
I wonder if we should throw an error if here if:  `!cmTransaction.minimumChargeAdjustment ` as that is the only type of transaction we would be expecting.

Reply:
Yes, good call!